src/blog/internet-computer-web-app-decentralized-database-architecture.md

```diff
@@ -0,0 +1,345 @@
+---
+path: "/blog/internet-computer-web-app-decentralized-database-architecture"
+date: "2021-08-12"
+title: "Internet Computer: Web App Decentralized Database Architecture"
+description: "An approach of the data persistence for web applications with the Internet Computer of the DFINITY foundation."
+tags: "#webdev #serverless #database #programming"
+image: "https://cdn-images-1.medium.com/max/2600/1*-Nq7tvQRO507Is_U_LSuMA.jpeg"
+canonical: "https://daviddalbusco.medium.com/singleton-factory-patterns-with-typescript-59e5a405531e"
+---
+![](https://cdn-images-1.medium.com/max/2600/1*-Nq7tvQRO507Is_U_LSuMA.jpeg)
+
+*Source: DFINITY [press kit](https://dfinity.org/media)*
+
+We are developing a proof of concept to port our web app, [DeckDeckGo](https://deckdeckgo.com/), to the Internet Computer of the [DFINITY](https://dfinity.org/) foundation.
+
+After having succeed validating the hosting and password-less authentication solutions, we are tackling the last question of our POC: persisting the data of our users, their presentations, in the blockchain network.
+
+Along the way, we tried out two concepts:
+
+* the “conservative one”:  a data persistence in a single database-like storage
+* the “futuristic one 🤯”: generate on the fly a decentralized secure data persistence (database-like) for each deck of each user.
+
+In this article I present these two approaches.
+
+*****
+
+### Introduction
+
+The scope of this blog post is limited to simple key-value database concepts. That kind of persistence is the one we rely on in DeckDeckGo.
+
+I do not know if anyone has yet implemented heavier concept with the IC (Internet Computer), such as a fully functional SQL database running in the chain.
+
+You can of course challenge these by commenting the blog post or by reaching out 😃.
+
+*****
+
+### The “Nowadays”
+
+Before digging into the Internet Computer, let's review first a common concept, a good old' school fashion way approach.
```

Comment:
```suggestion
Before digging into the Internet Computer, let's first review some common, old fashion concepts.
```

either "good old" (or "good ol'") or "old school" or "old fashion" but not all three 🤣